build/common/installer/scripts/tomlparser-prom-customconfig.rb

```diff
@@ -0,0 +1,423 @@
+    new_contents = new_contents.gsub("$AZMON_TELEGRAF_CUSTOM_PROM_KUBERNETES_FIELD_SELECTOR", "# Commenting this out since new plugins will be created per namespace\n  # $AZMON_TELEGRAF_CUSTOM_PROM_KUBERNETES_FIELD_SELECTOR")
+    new_contents = new_contents.gsub("$AZMON_TELEGRAF_CUSTOM_PROM_SCRAPE_SCOPE", "# Commenting this out since new plugins will be created per namespace\n  # $AZMON_TELEGRAF_CUSTOM_PROM_SCRAPE_SCOPE")
+
+    pluginConfigsWithNamespaces = ""
+    monitorKubernetesPodsNamespaces.each do |namespace|
+      if !namespace.nil?
+        #Stripping namespaces to remove leading and trailing whitespaces
+        namespace.strip!
+        if namespace.length > 0
+          pluginConfigsWithNamespaces += "\n[[inputs.prometheus]]
+  interval = \"#{interval}\"
+  monitor_kubernetes_pods = true
+  pod_scrape_scope = \"#{(@controller.casecmp(@replicaset) == 0) ? "cluster" : "node"}\"
+  monitor_kubernetes_pods_namespace = \"#{namespace}\"
+  kubernetes_label_selector = \"#{kubernetesLabelSelectors}\"
+  kubernetes_field_selector = \"#{kubernetesFieldSelectors}\"
+  fieldpass = #{fieldPassSetting}
+  fielddrop = #{fieldDropSetting}
+  metric_version = #{@metricVersion}
+  url_tag = \"#{@urlTag}\"
+  bearer_token = \"#{@bearerToken}\"
+  response_timeout = \"#{@responseTimeout}\"
+  tls_ca = \"#{@tlsCa}\"
+  insecure_skip_verify = #{@insecureSkipVerify}\n"
+        end
+      end
+    end
+    new_contents = new_contents.gsub("$AZMON_TELEGRAF_CUSTOM_PROM_PLUGINS_WITH_NAMESPACE_FILTER", pluginConfigsWithNamespaces)
+    return new_contents
+  rescue => errorStr
+    puts "Exception while creating prometheus input plugins to filter namespaces for custom prometheus: #{errorStr}, using defaults"
+    replaceDefaultMonitorPodSettings(new_contents, monitorKubernetesPods, kubernetesLabelSelectors, kubernetesFieldSelectors)
+  end
+end
+
+# Use the ruby structure created after config parsing to set the right values to be used as environment variables
+def populateSettingValuesFromConfigMap(parsedConfig)
+  if !@controller.nil?
+    if !parsedConfig.nil? && !parsedConfig[:prometheus_data_collection_settings].nil?
+      if @controller.casecmp(@replicaset) == 0 && !parsedConfig[:prometheus_data_collection_settings][:cluster].nil?
+        #Get prometheus replicaset custom config settings
+        begin
+          interval = parsedConfig[:prometheus_data_collection_settings][:cluster][:interval]
+          fieldPass = parsedConfig[:prometheus_data_collection_settings][:cluster][:fieldpass]
+          fieldDrop = parsedConfig[:prometheus_data_collection_settings][:cluster][:fielddrop]
+          urls = parsedConfig[:prometheus_data_collection_settings][:cluster][:urls]
+          kubernetesServices = parsedConfig[:prometheus_data_collection_settings][:cluster][:kubernetes_services]
+
+          # Remove below 4 lines after phased rollout
```

Comment:
What is the phased rollout?

Reply:
It is the rollout we are planning to make first to a few regions and then to all the regions. Have added you to the AKS PR.